### ISSUE TITLE
Aspect ratio correction when fullscreen is set

### ISSUE THREAD

**Issue by vanfanel** [ihhub/fheroes2#416]
Hi there again,

I have been looking at the code and I could not find any mention to aspect ratio correction, so the game runs "stretched" in fullscreen, which harms the (very beautiful) graphic art of this game.
Actual implementation would be as easy as doing:
`SDL_RenderSetLogicalSize(_renderer, 640, 480); `
just after SDL_CreateRenderer(), and an option for fheroes2.cfg would be needed too.
Should I go on, implement and do a PR for this small but important detail?

Thanks!

**Comment by ihhub** [comment]
Hi @vanfanel , if it applies only for fullscreen mode then surely it's a good to have feature. Please note that the original game allows to stretch a window in any aspect ratio in windowed mode so we should preserve such.

Moreover, if you're interested in fixing the resolution of the game I highly recommend to look into an implementation of dynamic resolution change within the game, not within the configuration file. If there is no such item in the issue list could you please create one as well?

Before going to implement these above enhancement our main concern is scaling within main screen (#286). If we fix that then we could go with the implementation of everything else without a fear that we have an ugly main screen for higher than 640x480 resolutions.

In conclusion, I give a decision to choose on what to work on for you :)

**Comment by vanfanel** [comment]
I have implemented a solution to this (and other SDL2-related options) in https://github.com/vanfanel/fheroes2/commit/ca70e5955f8a0210a78a9c1731b5866b24520bbe
